### ISSUE TITLE
Issues with the documentation

### ISSUE THREAD

**Issue by biprateep** [lincc-frameworks/python-project-template#217]
Hi,
While using the template and the documentation (https://lincc-ppt.readthedocs.io/en/latest/source/new_project.html), I faced a couple of difficulties:
- `pip install -e .` installs an editable version of the package i.e. if I change the source and reload it uses the new version but doing `pip install -e '.[dev]' ` makes the installed package non-editable again.
- If you name your initial local branch `initial_branch` and try to push it to the empty GitHub repo, then it creates two diverging histories and is problematic to merge.

**Comment by drewoldag** [comment]
Thanks for the feedback @biprateep! 

@cchris28 you recently made the changes to the docs in response to issue #203, do you have thoughts on the point brought up here?

Similarly, I have also encountered the `initial_branch` and divergent histories issue. I'm curious if it makes sense to maintain the pre-commit hook that prevents merging to main? 

I know there is a setting in github that will prevent users from committing to main directly. Perhaps we encourage people to use that approach instead of specifying an approach that requires the complicated `initial_branch` and resolution of diverging histories.

**Comment by cchris28** [comment]
My experience differs in that I find that "pip install -e .[dev]" does install an editable package. I am unclear on the steps to reproduce the problem bitrapeep reported. The reason I suggested adding the -e option to the prior "pip install .[dev]" was because THAT made the package not editable.

I am inclined to get rid of the refusal to check in to main. It does interact badly with the git histories as reported.

**Comment by drewoldag** [comment]
@biprateep I took a swing at trying to reproduce the issue you were seeing with the `pip install -e .; pip install -e .[dev]` commands, and wasn't able to get the same results in my local instance. 

If you've got any additional details of how you set up your environment, that you can share, I can try to take another look.

**Comment by biprateep** [comment]
Hi @drewoldag and @cchris28 ,
Thanks for looking into this. I just realized what has happened. I encountered the issue with the editability of the repo some time ago and between the time it took me actually to post the issue, you had updated the documentation. If I follow the updated guidelines the issue is not present. I did not realize that a change has happened and naively posted about it here. Apologies for that. Though I believe the issue with diverging git histories has not been addressed yet.
